code/go/0chain.net/smartcontract/storagesc/clientpool.go

```diff
@@ -0,0 +1,722 @@
+package storagesc
+
+import (
+	"0chain.net/smartcontract"
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/url"
+
+	cstate "0chain.net/chaincore/chain/state"
+	"0chain.net/chaincore/state"
+	"0chain.net/chaincore/transaction"
+	"0chain.net/core/common"
+	"0chain.net/core/datastore"
+	"0chain.net/core/util"
+)
+
+//
+// client read/write pool
+//
+
+func clientPoolKey(scKey, clientID string) datastore.Key {
+	return datastore.Key(scKey + ":clientpool:" + clientID)
+}
+
+type clientPool struct {
+	Allocations allocationPools `json:"allocations"`
+}
+
+// Encode pool
+// Implements util.Serializable interface.
+func (cPool *clientPool) Encode() []byte {
+	var b, err = json.Marshal(cPool)
+	if err != nil {
+		panic(err) // must never happens
+	}
+	return b
+}
+
+// Decode pool
+// Implements util.Serializable interface
+func (cPool *clientPool) Decode(p []byte) error {
+	return json.Unmarshal(p, cPool)
+}
+
+// Save pool in tree
+func (cPool *clientPool) save(ssContractKey, clientID string, balances cstate.StateContextI,
+) (err error) {
+	_, err = balances.InsertTrieNode(clientPoolKey(ssContractKey, clientID), cPool)
+	return
+}
+
+// todo: add description, rename it
+func (cPool *clientPool) blobberCut(aPoolID, bPoolID string, eTimestamp common.Timestamp,
```

Comment:
Its just a wrapper around a blobberCut method so no reason to name it anything else, unless you rename the whole stack.